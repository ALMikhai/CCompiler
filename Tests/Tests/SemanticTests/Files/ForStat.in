int f() {
    int a = 1;
    int i = 0;
    for (;;)
        a = 2;
        
    for (i = 0;;) 
        a = 2;
    
    for (i = 0; i < 10;) 
        a = 2;
        
    for (i = 0; i < 10; ++i) 
        a = 2;
}