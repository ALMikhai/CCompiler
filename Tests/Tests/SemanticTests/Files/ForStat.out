\-TRANSLATIONUNIT
  \-FUNC
    |-DECLSPEC
    | |-INT
    |-INITDECLARATOR
    | \-DECLARATOR
    |   \-FUNC
    |     |-f
    |     \-EMPTY
    \-COMPOUNDSTAT
      |-DECLLIST
      | |-DECL
      | | |-DECLSPEC
      | | | |-INT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-a
      | |     \-1
      | \-DECL
      |   |-DECLSPEC
      |   | |-INT
      |   \-INITDECLARATORLIST
      |     \-INITDECLARATOR
      |       |-DECLARATOR
      |       | \-i
      |       \-0
      \-STATLIST
        |-FOR
        | |-EMPTY
        | |-EMPTY
        | |-EMPTY
        | \-ASSIGN
        |   |-a
        |   \-2
        |-FOR
        | |-ASSIGN
        | | |-i
        | | \-0
        | |-EMPTY
        | |-EMPTY
        | \-ASSIGN
        |   |-a
        |   \-2
        |-FOR
        | |-ASSIGN
        | | |-i
        | | \-0
        | |-LESS
        | | |-i
        | | \-10
        | |-EMPTY
        | \-ASSIGN
        |   |-a
        |   \-2
        \-FOR
          |-ASSIGN
          | |-i
          | \-0
          |-LESS
          | |-i
          | \-10
          |-PREFIX INC
          | \-i
          \-ASSIGN
            |-a
            \-2

Structs { }
Symbols {
	FUNC returning VOID
	Arguments{
		STRING :: s
		INT :: o
	}:: printf
	FUNC returning INT
	Arguments{ }
	Structs { }
	Symbols {
		INT :: a
		INT :: i
	} :: f
}
